### PR TITLE
support salvo build from ci container

### PR DESCRIPTION
**Pull request by xu1zhou**
support salvo build from ci container.

Works on #150.

### CODE REVIEW

**Review by mum4k — CHANGES_REQUESTED** [review]
Thank you for improving the build script.

File: ci/run_envoy_docker.sh [hunk 1]
```diff
@@ -38,14 +38,14 @@ else
...
   ENVOY_DOCKER_OPTIONS+=(--cap-add SYS_PTRACE --cap-add NET_RAW --cap-add NET_ADMIN)
   DEFAULT_ENVOY_DOCKER_BUILD_DIR=/tmp/envoy-docker-build
-  BUILD_DIR_MOUNT_DEST=/build
+  [[ -z "${BUILD_DIR_MOUNT_DEST}" ]] && BUILD_DIR_MOUNT_DEST="/build" # Salvo unique
   SOURCE_DIR="${PWD}"
-  SOURCE_DIR_MOUNT_DEST=/source
+  [[ -z "${SOURCE_DIR_MOUNT_DEST}" ]] && SOURCE_DIR_MOUNT_DEST="/source" # Salvo unique
   START_COMMAND=("/bin/bash" "-lc" "groupadd --gid $(id -g) -f envoygroup \
-    && useradd -o --uid $(id -u) --gid $(id -g) --no-create-home --home-dir /build envoybuild \
+    && useradd -o --uid $(id -u) --gid $(id -g) --no-create-home --home-dir ${BUILD_DIR_MOUNT_DEST} envoybuild \
```
Comment (gyohuangxin) [inline ci/run_envoy_docker.sh:45]: The comment `# Salvo unique` is required in each changed line.

File: ci/run_envoy_docker.sh [hunk 2]
```diff
@@ -38,14 +38,14 @@ else
...
     && usermod -a -G pcap envoybuild \
-    && chown envoybuild:envoygroup /build \
-    && sudo -EHs -u envoybuild bash -c 'cd /source && $*'")
+    && chown envoybuild:envoygroup ${BUILD_DIR_MOUNT_DEST} \
```
Comment (gyohuangxin) [inline ci/run_envoy_docker.sh:47]: Ditto

File: salvo/README.md [hunk 3]
```diff
@@ -31,6 +31,12 @@ To build Salvo, use the following command:
 bazel build //...
 ```
 
+Salvo also support building by docker, use the following command:
```
Comment (gyohuangxin) [inline salvo/README.md:34]:
```suggestion
Salvo also supports building by docker, use the following command:
```

File: salvo/README.md [hunk 4]
```diff
@@ -31,6 +31,12 @@ To build Salvo, use the following command:
...
+
+```bash
+ENVOY_DOCKER_BUILD_DIR=/tmp/salvo-docker-build BUILD_DIR_MOUNT_DEST=/tmp/salvo-docker-build SOURCE_DIR_MOUNT_DEST=path_of_envoy-perf   ci/run_envoy_docker.sh 'ci/do_ci.sh build'
```
[the remark this answers is not in the document]
Comment (xu1zhou) [reply salvo/README.md]: sure

File: ci/run_envoy_docker.sh (hunk 1 above)
Comment (xu1zhou) [reply ci/run_envoy_docker.sh:45]: add comment behind a backslash for each line  in a multiline command will break this command. I take an alternative way  as below 
```
echo CommandName InputFiles `#1st comment` \
             --option1 arg1 `#2nd comment` \
             --option2 arg2 `#3rd comment`
```
it's not align with other comment style but it's the only way i found by now.

Comment (gyohuangxin) [reply ci/run_envoy_docker.sh:45]: It works for me, the comment `#Salvo unique` is important for maintainers to figure out the difference from Nighthawk and Envoy scripts.

**Review by gyohuangxin — APPROVED** [review]
LGTM, thanks for improving Salvo continually.

**Pull request merged**